js/digifinex.js

```diff
@@ -0,0 +1,528 @@
+const { AccountSuspended, BadRequest, BadResponse, NetworkError, DDoSProtection, AuthenticationError, PermissionDenied, ArgumentsRequired, ExchangeError, InsufficientFunds, InvalidOrder, InvalidNonce } = require ('./base/errors');
+const { DECIMAL_PLACES } = require ('./base/functions/number');
+// const log = require ('ololog');
+
+//  ---------------------------------------------------------------------------
+
+module.exports = class digifinex extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'digifinex',
+            'name': 'DigiFinex',
+            'countries': [ 'SG' ],
+            'version': 'v1',
+            'rateLimit': 900, // 300 for posts
+            'certified': false,
+            // new metainfo interface
+            'has': {
+                // 'cancelAllOrders': false,
+                'cancelOrders': true,
+                'createMarketOrder': false,
+                'fetchBalance': true,
+                'fetchClosedOrders': true,
+                'fetchTradingFee': false,
+                'fetchTradingFees': false,
+                'fetchFundingFees': false,
+                // 'fetchMyTrades': true,
+                'fetchOHLCV': true,
+                'fetchOpenOrders': true,
+                'fetchOrder': true,
+                'fetchTickers': true,
+            },
+            'timeframes': {
+                '1m': '1m',
+                '5m': '5m',
+                '15m': '15m',
+                '30m': '30m',
+                '1h': '1h',
+                '4h': '4h',
+                '12h': '12h',
+                '1d': '1d',
+                '1w': '1w',
+            },
+            'urls': {
+                'logo': 'https://static.digifinex.vip/newhome/pc/img/index/logo_dark.svg',
+                'api': 'https://openapi.digifinex.vip/v2',
+                'www': 'https://www.digifinex.vip/',
+                'doc': [
+                    'https://github.com/DigiFinex/api',
+                ],
+                'fees': 'https://digifinex.zendesk.com/hc/en-us/articles/360007166473-Fee-Structure-on-DigiFinex',
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'ticker',
+                    ],
+                },
+                'private': {
+                    'get': [
+                        'otc_market_price',
+                        'depth',
+                        'trade_detail',
+                        'kline',
+                        'trade_pairs',
+                        'open_orders',
+                        'order_history',
+                        'order_info',
+                        'order_detail',
+                        'myposition',
+                    ],
+                    'post': [
+                        'trade',
+                        'cancel_order',
+                    ],
+                },
+            },
+            'commonCurrencies': {
```

Comment:
This mapping should be removed, because we only need to have it if it converts one currency code to another currency code, in this case `commonCurrencies` does no conversions or unifications as [explained in the Manual](https://github.com/ccxt/ccxt/wiki/Manual#symbols-and-market-ids), so we can simply drop it from here.

Reply:
Still not quite sure what it is for. I added DFT => DigiFinexToken. Is that alright?

Reply:
> Still not quite sure what it is for

It is needed in cases where the exchange names the currency `XBT` meaning `BTC`. Or names it `DSH`, meaning `DASH`. Or in other such cases, where a renaming is needed.

However, for DFT – we don't need to rename it to DigiFinexToken, we would rather keep it as DFT everywhere.

So, in this particular case, the `.commonCurrencies` aren't needed at all and can be dropped.